src/Domain/Person.php

```diff
@@ -140,49 +175,68 @@ public function getLastName(): string
 
     public function hashPassword(): void
     {
-        $this->password = Password::hash($this->raw_password);
+        if (!empty($this->raw_password)) {
+            $this->password = Password::hash($this->raw_password);
+        }
     }
 
-    public function getPasswordHash(): string
+    public function getPasswordHash(): ?string
     {
         return $this->password;
     }
 
+    public function setStripeCustomerId(?string $stripe_customer_id): void
+    {
+        $this->stripe_customer_id = $stripe_customer_id;
+    }
+
     #[\ReturnTypeWillChange]
     public function jsonSerialize(): array
     {
+        $this->has_password = $this->getPasswordHash() !== null;
+
         $jsonVars = get_object_vars($this);
         $jsonVars['uuid'] = $this->getId()?->toString();
 
         return $jsonVars;
     }
 
     /**
-     * @Assert\Callback()
+     * @Assert\Callback(groups={"new"})
      * @see Person::$captcha_code
-     * @see Person::$raw_password
      */
-    public function validateCaptchaAndRawPasswordSetIfNew(ExecutionContextInterface $context): void
+    public function validateCaptchaIfNew(ExecutionContextInterface $context): void
```

Comment:
What happens if I send a dummy captcha? I presume this is to be implemented later?

Reply:
@alihejazi No, the middleware already checks that it is actually a solved captcha. I will update this method name to be clearer.